internal/ingress/controller/store/store_test.go

```diff
@@ -168,21 +158,22 @@ func TestStore(t *testing.T) {
 
 		storer.Run(stopCh)
 
-		ing, err := ensureIngress(&v1beta1.Ingress{
+		ing := ensureIngress(&extensions.Ingress{
 			ObjectMeta: metav1.ObjectMeta{
 				Name:      "dummy",
 				Namespace: ns,
+				SelfLink:  fmt.Sprintf("/apis/extensions/v1beta1/namespaces/%s/ingresses/dummy", ns),
```

Comment:
Isn't this field generated by the API server, and not the user?

Reply:
I don't think the fakeClient mocks generating this field, when it is not provided in the ingress the following is obtained:

```
Could not construct reference to: '<ingress-resource>' due to: 'selfLink was empty, can't make reference'
```